src/popup/functions/updateOptions.js

```diff
@@ -5,8 +5,20 @@ import updateSliderKnob from './updateSliderKnob.js';
  * @param result {object}
  */
 export default function (result) {
-  document.querySelector(`#${result.optOut.selector}`).checked = true;
   const slider = document.querySelector('#slider');
-  slider.value = result.optOut.slider;
-  updateSliderKnob(slider);
+  // If no saved settings in local storage
```

Comment:
This if statement can be done in a simpler & slightly more reliable way:

```js
if(result && result.optOut && result.optOut.slider && result.optOut.selector) {
    // If there are saved settings
    document.querySelector(`#${result.optOut.selector}`).checked = true;
    // Set the settings on the slider
    slider.value = result.optOut.slider;
    updateSliderKnob(slider);
} else {
 // Set slider class for display
    updateSliderKnob(slider);
}
```

You can find more info here on why checking with `== null` can lead to unexpected bugs. Best is to always use `===` when comparing to values, and only test for truthy/falsy to check if something exists:

https://guide.freecodecamp.org/javascript/falsy-values/

Reply:
I used `== null` to check if it's either null or undefined. And think it risks introducing more bugs to risk checking for truthiness in vanilla JS.

Reply:
Rather than checking truthiness I prefer to put everything in to an explicit boolean comperator so that we always at least know the type returned from the operation. But I program in typescript mostly so truthiness is best avoided.

Reply:
Since it depends on browser.storage.sync.get result can only be defined value, undefined, or throw and error.